tests/ccl_test_pyccl_interface.py

```diff
@@ -732,8 +784,8 @@ def test_valid_transfer_combos():
     
     assert_raises(ValueError, ccl.Cosmology, transfer_function='emulator', 
                               matter_power_spectrum='linear', **params)
-    assert_raises(ValueError, ccl.Cosmology, transfer_function='boltzmann', 
-                              matter_power_spectrum='halomodel', **params)
+    #assert_raises(ValueError, ccl.Cosmology, transfer_function='boltzmann', 
```

Comment:
Why commented out?

Reply:
Okay,  a bit of a long story here:

First; previously there were `halo_model` and `halomodel` cases for `ccl_nonlin_matter_power`, neither of which were wired up to anything. If you called `ccl_nonlin_matter_power` with each method I think you just got the linear power back. I removed one of them so we are now left with `halo_model` only. I couldn't think of any good reason to have both.

Second; I looked at wiring the `halo_model` option up to the new calculation from this new halomod branch, but this is impossible I think because `ccl_halomod.c` uses `ccl_power.h`, so I can't use the correct routines without entering some sort of infinite calling loop. So at the moment the `halo_model` option remains useless.

This sort of relates to one of the issues I raised previously (https://github.com/LSSTDESC/CCL/issues/337) and also sort of to https://github.com/LSSTDESC/CCL/issues/438

Generally, I think it would be better to have a source file dedicated to getting the linear power spectrum (`ccl_linear_power.c` ?) and then a separate file(s) dedicated to whatever non-linear method you wanted to use... and then one final source file that had a way of choosing between whatever power spectrum calculation the user needed, be that linear, emu, halomodel, halofit... whatever.